src/SearchQueryService/Indexes/IndexesProcessor.cs

```diff
@@ -33,10 +43,35 @@ public async Task ProcessDirectory()
                 var postBody = CreateSchemaPostBody(fieldsToAdd);
                 CreateCoreSchema(postBody, index.Name);
 
+                await WaitUntilSchemaCreated(4, 500, fieldsToAdd.Count(), index.Name);
```

Comment:
Voláš tu metódu `.Count()`, ktorá začne opätovne iterovať cez daný `IEnumerator`, ktorý je v metóde definovaný dosť komplexne. Preto to nie je šťastné. Na riadku `42` si radšej zavolaj `ToList()` aby si to mal v liste a potom môžeš použiť property `.Count`.

Reply:
nie je ale `.ToList()` ešte pomalšie ako `.Count()`? Ďalej už s Enumeratorom robím len `.Where().Select()` a to by malo byť rovnako rýchle pre List, aj Enumerator.